docker/server/entrypoint.sh

```diff
@@ -3,164 +3,213 @@
 set -eo pipefail
 shopt -s nullglob
 
-DO_CHOWN=1
-if [ "${CLICKHOUSE_DO_NOT_CHOWN:-0}" = "1" ]; then
-    DO_CHOWN=0
-fi
+# check to see if this file is being run or sourced from another script
+_is_sourced() {
+	# https://unix.stackexchange.com/a/215279
+	[ "${#FUNCNAME[@]}" -ge 2 ] \
+		&& [ "${FUNCNAME[0]}" = '_is_sourced' ] \
+		&& [ "${FUNCNAME[1]}" = 'source' ]
+}
+
+get_variables() {
+    declare -g DO_CHOWN
+    DO_CHOWN=1
+    if [ "${CLICKHOUSE_DO_NOT_CHOWN:-0}" = "1" ]; then
+        DO_CHOWN=0
+    fi
 
-CLICKHOUSE_UID="${CLICKHOUSE_UID:-"$(id -u clickhouse)"}"
-CLICKHOUSE_GID="${CLICKHOUSE_GID:-"$(id -g clickhouse)"}"
-
-# support --user
-if [ "$(id -u)" = "0" ]; then
-    USER=$CLICKHOUSE_UID
-    GROUP=$CLICKHOUSE_GID
-    if command -v gosu &> /dev/null; then
-        gosu="gosu $USER:$GROUP"
-    elif command -v su-exec &> /dev/null; then
-        gosu="su-exec $USER:$GROUP"
+    declare -g CLICKHOUSE_UID CLICKHOUSE_GID
+    CLICKHOUSE_UID="${CLICKHOUSE_UID:-"$(id -u clickhouse)"}"
+    CLICKHOUSE_GID="${CLICKHOUSE_GID:-"$(id -g clickhouse)"}"
+
+    declare -g gosu USER GROUP
+    # support --user
+    if [ "$(id -u)" = "0" ]; then
+        USER=$CLICKHOUSE_UID
+        GROUP=$CLICKHOUSE_GID
+        if command -v gosu &> /dev/null; then
+            gosu="gosu $USER:$GROUP"
+        elif command -v su-exec &> /dev/null; then
+            gosu="su-exec $USER:$GROUP"
+        else
+            echo "No gosu/su-exec detected!"
+            exit 1
+        fi
     else
-        echo "No gosu/su-exec detected!"
-        exit 1
+        # User needs to exist in docker image
+        # build a new image with the user desired
+        USER="$(id -u)"
+        GROUP="$(id -g)"
+        gosu=""
+        DO_CHOWN=0
     fi
-else
-    USER="$(id -u)"
-    GROUP="$(id -g)"
-    gosu=""
-    DO_CHOWN=0
-fi
-
-# set some vars
-CLICKHOUSE_CONFIG="${CLICKHOUSE_CONFIG:-/etc/clickhouse-server/config.xml}"
 
-if ! $gosu test -f "$CLICKHOUSE_CONFIG" -a -r "$CLICKHOUSE_CONFIG"; then
-    echo "Configuration file '$dir' isn't readable by user with id '$USER'"
-    exit 1
-fi
+    declare -g CLICKHOUSE_CONFIG SERVER_ARGS
+    CLICKHOUSE_CONFIG="${CLICKHOUSE_CONFIG:-/etc/clickhouse-server/config.xml}"
+    SERVER_ARGS="$SERVER_ARGS --config-file=$CLICKHOUSE_CONFIG"
```

Comment:
Config path can (theoretically) have spaces. Pass with quotes

Reply:
I've added double quotes on:

`exec $gosu "$@" "$SERVER_ARGS"`

i think now it's ok.